File: c/collatz-conjecture/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start)
{
	if (start < 1)
		return ERROR_VALUE;

	int steps = 0;

	while (start > 1)
	{
		if (start % 2 == 0)
			start /= 2;
		else
			start = start * 3 + 1;
		steps++;
	}

	return steps;
}

File: go/collatz-conjecture/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (steps int, err error) {
	if n < 1 {
		return -1, errors.New("invalid input, must be greater than 0")
	}

	for ; n > 1; steps++ {
		if n%2 == 0 {
			n /= 2
		} else {
			n = n*3 + 1
		}
	}

	return
}
